### PR TITLE
EB Platform Hook - LMS Via

### PR DESCRIPTION
**Pull request by indigobravo**
Created the 10_set-env-var.sh platform hook. The script is executed
during the EB `predeploy` phase, and is used to set a long environment
variable.

This works around a limitation on the size of an environment variable
imposed by elastic beanstalk.

### CODE REVIEW

File: lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+#
+# 10_set-env-var.sh
+#
+# Custom script to work around a limitation on the size of an
+# environment variable imposed by elastic beanstalk.
```
Comment (robertknight) [inline lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh:6]:
```suggestion
# environment variable imposed by Elastic Beanstalk.
```

Pardon my pedantry.

File: lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+set -eu
+
+env_file="/opt/elasticbeanstalk/deployment/env.list"
+
+google_service_account=$(aws ssm get-parameter \
+  --with-decryption \
+  --name eb-lms-via-prod-gsa \
+  --region us-west-1 \
+  | jq -r .Parameter.Value
+  )
```
Comment (robertknight) [inline lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh:21]: Where does the `aws` command gets its authorization from? Is it based on the identity of the calling machine or something else?

File: lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+#
+# 10_set-env-var.sh
+#
+# Custom script to work around a limitation on the size of an
+# environment variable imposed by elastic beanstalk.
+#
+# We are taking a value from the SSM parameter store and adding it to
+# the `env.list` file which gets used to set Docker env vars at
+# runtime.
```
Comment (robertknight) [inline lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh:10]: Helpful comment. I guessed as much from the code but it is helpful to clarify.

**Review by robertknight — APPROVED** [review]
This looks nice and simple 👍

File: lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh (hunk 2 above)
Comment (indigobravo) [reply lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh:21]: Correct - The EC2 servers running the service have the ability to `get` a parameter from the SSM parameter store.

File: lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh (hunk 1 above)
Comment (indigobravo) [reply lms-via/platform/prod/hooks/predeploy/10_set-env-var.sh:6]: Thanks - I will update.

**Pull request merged**